python/pyarrow/_dataset.pyx

```diff
@@ -2838,6 +2838,13 @@ cdef class Scanner(_Weakrefable):
             result = self.scanner.Head(num_rows)
         return pyarrow_wrap_table(GetResultValue(result))
 
+    def count_rows(self):
+        """Count rows matching the scanner filter."""
```

Comment:
I was going to comment that adding a similar function to `Fragment` might also be nice, but I assume the user can now just do `fragment.scanner().count_rows()` instead of `fragment.count_rows()` ?

Reply:
They can, yes. However I could add the helper function to Fragment for consistency (and I think there are a number of other helper functions that I could also add).